### DISCUSSION TITLE
Api connection

### DISCUSSION THREAD

**Discussion by ghost** [albar965/littlenavmap#1186] [Q&A] [answered]
Script started
2024-09-05 21:38:07,503 - INFO - Bot instance created
2024-09-05 21:38:07,505 - INFO - Attempting to connect to LittleNavMap...
Failed to retrieve data: 400, Content: Controller not found
2024-09-05 21:38:07,509 - WARNING - Failed to retrieve data from LittleNavMap.
2024-09-05 21:38:07,510 - INFO - Indexes created on conversation collection.
2024-09-05 21:38:09,107 - INFO - Bot is ready. Logged in as | your_ai_overlord
2024-09-05 21:38:09,107 - INFO - Attempting to connect to Speaker.bot at ws://localhost:7580, attempt 1
2024-09-05 21:38:11,141 - INFO - Successfully connected to Speaker.bot
2024-09-05 21:38:11,141 - DEBUG - All initial tasks created successfully
2024-09-05 21:38:12,541 - INFO - Listening for voice commands...

So I've written a python twitch bot but can't seem to get what I need from LittleNavMap. Controller not found, can anyone shed any light?

**Comment by albar965** [comment]
Sorry, but your report lacks all kind of information.

What URLs do you use to access the LNM web API (I guess) and what do you expect to get?
LNM log file?

Alex

**Comment by ghost** [comment]
LITTLENAVMAP_API_URL='http://localhost:8965'
[abarthel-little_navmap.log](https://github.com/user-attachments/files/16907651/abarthel-little_navmap.log)

I'm hoping to extract aircraft info, positional, speeds, etc.

Cheers for the prompt reply!

Matt

**Reply by albar965** [reply, marked as answer]
> LITTLENAVMAP_API_URL='http://localhost:8965'

The base URL does not give you anything. Did you look at [webapi](https://www.littlenavmap.org/manuals/littlenavmap/release/latest/en/WEBSERVER.html#webapi) or [webapi.yaml](https://github.com/albar965/littlenavmap/blob/release/3.0/web/webapi.yaml)?

Unfortunately the log does not help much since I removed the logging of requests due to flooding the log file. Completely forgot that.
You can enable detailed logging by changing `WebserverDebug=false` to `WebserverDebug=true` in section `[Options]` of the settings file [little_navmap.ini](https://www.littlenavmap.org/manuals/littlenavmap/release/latest/en/FILES.html#settings-directory).

**Reply by ghost** [reply]
Will do, cheers!

**Comment by ghost** [comment]
https://github.com/Matt-Cummins/Twitch-Flight-Simulation-Bot

Managed to get info from LNM for sim independent Twitch bot. Thanks so much for making this program!

**Reply by albar965** [reply]
Wow! Looks like a fun project although I could not test it. Like the sarcastic personality trait. :upside_down_face: 

I suppose you use the [Web API](https://github.com/albar965/littlenavmap/blob/release/3.0/web/webapi.yaml) of LNM. Just to be sure I don't break anything when changing the HTML output since some parse the HTML.

**Reply by albar965** [reply]
I added your project to https://albar965.github.io/links.html . Hope it's ok that I copied your text.
